sys/net/routing/ng_rpl/of0.c

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (C) 2014 Oliver Hahm <oliver.hahm@inria.fr>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     net_ng_rpl
+ * @{
+ * @file        of0.c
+ * @brief       Objective Function Zero.
+ *
+ * Implementation of Objective Function Zero.
+ *
+ * @author      Eric Engel <eric.engel@fu-berlin.de>
+ * @}
+ */
+
+#include <string.h>
+#include "of0.h"
+#include "net/ng_rpl.h"
+#include "net/ng_rpl/structs.h"
+
+//Function Prototypes
+static uint16_t calc_rank(ng_rpl_parent_t *, uint16_t);
+static ng_rpl_parent_t *which_parent(ng_rpl_parent_t *, ng_rpl_parent_t *);
+static ng_rpl_dodag_t *which_dodag(ng_rpl_dodag_t *, ng_rpl_dodag_t *);
+static void reset(ng_rpl_dodag_t *);
+
+static ng_rpl_of_t ng_rpl_of0 = {
+    0x0,
+    calc_rank,
+    which_parent,
+    which_dodag,
+    reset,
+    NULL,
+    NULL,
+    NULL
+};
+
+ng_rpl_of_t *ng_rpl_get_of0(void)
+{
+    return &ng_rpl_of0;
+}
+
+void reset(ng_rpl_dodag_t *dodag)
+{
+    /* Nothing to do in OF0 */
+    (void) dodag;
+}
+
+uint16_t calc_rank(ng_rpl_parent_t *parent, uint16_t base_rank)
+{
+    if (base_rank == 0) {
+        if (parent == NULL) {
```

Comment:
`base_rank = parent->rank` belongs to the outer `if` context. While merging both `ifs` would reduce the indentation, another if condition would be needed to check again for `base_rank == 0`